index.js

```diff
@@ -1,34 +1,13 @@
 'use strict'
 
 const fp = require('fastify-plugin')
-const mysql = require('mysql')
 
 function fastifyMysql (fastify, options, next) {
-  const name = options.name
-  delete options.name
-
-  const pool = mysql.createPool(options.connectionString || options)
-  const db = {
-    connect: onConnect => pool.getConnection(onConnect),
-    pool: pool,
-    query: pool.query.bind(pool)
-  }
-
-  if (name) {
-    if (!fastify.mysql) {
-      fastify.decorate('mysql', {})
-    }
-
-    fastify.mysql[name] = db
+  if ((options.type || 'pool') === 'pool') {
+    fastify.register(require('./pool'), options)
   } else {
-    if (fastify.mysql) {
-      next(new Error('fastify-mysql has already registered'))
-    } else {
-      fastify.mysql = db
-    }
+    fastify.register(require('./connection'), options)
```

Comment:
there seem to be no `connection.js` file.

Reply:
Not finished yet, it's a proposal. In most case, pool is more convenient and powerful. If we just want a connection, can set connectionLimit=1, or call pool.getConnection to get.
So, not sure should we separate pool/connection ?